src/api/app/controllers/webui/webui_controller.rb

```diff
@@ -88,6 +88,8 @@ class MissingParameterError < RuntimeError; end
   end
 
   def set_project
+    # We've started to use project_name for new routes...
+    params[:project] = params[:project_name] if params[:project_name].present?
```

Comment:
I think you could avoid overwriting the param by writing
```ruby
@project = Project.find_by(name: params[:project_name] || params[:project])
```
and changing the flash in line 14 to use @project.name.

I can't say right now if it causes an XSS, but I would prefer to avoid overwriting params if possible.